cmd/mdatagen/internal/metadata.go

```diff
@@ -338,3 +343,52 @@ func (a Attribute) TestValue() string {
 	}
 	return ""
 }
+
+func (a Attribute) TestValueTemplateKey() string {
+	switch a.Type.ValueType {
+	case pcommon.ValueTypeEmpty:
+		return ""
+	case pcommon.ValueTypeStr:
+		return `"key", "val"`
+	case pcommon.ValueTypeInt:
+		return `"key", 1`
+	case pcommon.ValueTypeDouble:
+		return `"key", 1.1`
+	case pcommon.ValueTypeBool:
+		return `"key", true`
+	case pcommon.ValueTypeMap:
+		return `"key", map[string]any{"key1": "val1", "key2": "val2"}`
+	case pcommon.ValueTypeSlice:
+		return `"key", []any{"item1", "item2"}`
+	case pcommon.ValueTypeBytes:
+		return `"key", []byte("val")`
+	}
+	return ""
+}
+
+func (a Attribute) TestResultTemplateValue() string {
```

Comment:
```suggestion
func (a Attribute) TestTemplateValue() string {
```
Minor, but we remove `Result` here? It's a little unclear to me what is intended given `TestValue` doesn't have it, and the only difference in this function is that it only handles possible template value types. If we want to keep it, I think a brief comment might be helpful.